tests/FSharp.Test.Utilities/CompilerAssert.fs

```diff
@@ -83,19 +83,32 @@ type CompilerAssert private () =
 
     static let _ = config |> ignore
 
-// Do a one time dotnet sdk build to compute the proper set of reference assemblies to pass to the compiler
-#if NETCOREAPP
+    // Do a one time dotnet sdk build to compute the proper set of reference assemblies to pass to the compiler
     static let projectFile = """
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>$TARGETFRAMEWORK</TargetFramework>
     <UseFSharpPreview>true</UseFSharpPreview>
     <DisableImplicitFSharpCoreReference>true</DisableImplicitFSharpCoreReference>
   </PropertyGroup>
 
   <ItemGroup><Compile Include="Program.fs" /></ItemGroup>
+  <ItemGroup><Reference Include="$FSHARPCORELOCATION" /></ItemGroup>
+  <ItemGroup Condition="'$(TARGETFRAMEWORK)'=='net472'">
+    <Reference Include="System" />
+    <Reference Include="System.Core.dll" />
+    <Reference Include="System.Xml.Linq.dll" />
+    <Reference Include="System.Data.DataSetExtensions.dll" />
+    <Reference Include="Microsoft.CSharp.dll" />
+    <Reference Include="System.Data.dll" />
+    <Reference Include="System.Deployment.dll" />
+    <Reference Include="System.Drawing.dll" />
+    <Reference Include="System.Net.Http.dll" />
+    <Reference Include="System.Windows.Forms.dll" />
+    <Reference Include="System.Xml.dll" />
+  </ItemGroup>
```

Comment:
Why this instead of multi-target? Shouldn't these tests just be netcoreapp3.1 anyways?

Reply:
This is just the code to get the list of assemblies for the test harness to get for references for the build.  So we do it once per test run.